File: src/dashboard/data/pages.js
export const sidebarLinks = [
  { path: 'overview', icon: 'fa-chart-line', label: 'Overview', id: 'Overview' },
  { path: 'servers/servers', icon: 'fa-server', label: 'Servers', id: 'Servers' },
  { path: 'tickets/tickets', icon: 'fa-ticket', label: 'Tickets', id: 'Tickets' },
  { path: 'giveaways/giveaway', icon: 'fa-gift', label: 'Giveaways', id: 'Giveaways' },
  { path: 'moderation/moderation', icon: 'fa-shield', label: 'Moderation', id: 'Moderation' },
  { path: 'dashboard', icon: 'fa-gear', label: 'Settings', id: 'Dashboard' },
  { path: 'leveling/leveling', icon: 'fa-ranking-star', label: 'Leveling', id: 'Leveling' },
];

const overviewRightNav = [
  { path: 'overview/statistics', icon: 'fa-chart-bar', label: 'Statistics', id: 'Statistics' },
  { path: 'overview/settings', icon: 'fa-cog', label: 'Settings', id: 'Settings' },
  { path: 'overview/alerts', icon: 'fa-bell', label: 'Alerts', id: 'Alerts' },
  { path: 'overview/history', icon: 'fa-history', label: 'History', id: 'History' },
];

const serversRightNav = [
  { path: 'servers/add-server', icon: 'fa-plus', label: 'Add Server', id: 'AddServer' },
  { path: 'servers/manage-roles', icon: 'fa-users-cog', label: 'Manage Roles', id: 'ManageRoles' },
  { path: 'servers/permissions', icon: 'fa-key', label: 'Permissions', id: 'Permissions' },
  { path: 'servers/backup', icon: 'fa-save', label: 'Backup', id: 'Backup' },
];

const ticketsRightNav = [
  { path: 'tickets/tickets', icon: 'fa-gears', label: 'Settings', id: 'Settings' },
  { path: 'tickets/ticket-embed', icon: 'fa-regular fa-message', label: 'Embed', id: 'TicketEmbed' },
  { path: 'tickets/ticket-staff', icon: 'fa-people-group', label: 'Ticket Staff', id: 'TicketStaff' },
  { path: 'tickets/transcripts', icon: 'fa-file', label: 'Transcripts', id: 'Transcripts' },
  { path: 'tickets/panels', icon: 'fa-arrow-pointer', label: 'Panels', id: 'Panels' },
  { path: 'tickets/forms', icon: 'fa-file-lines', label: 'Forms', id: 'Forms' },
  { path: 'tickets/tags', icon: 'fa-tag', label: 'Tags', id: 'Tags' },
  { path: 'tickets/audit-logs', icon: 'fa-file-medical', label: 'Audit logs', id: 'AuditLogs' },
];

const giveawaysRightNav = [
  { path: 'giveaways/create-giveaway', icon: 'fa-plus-circle', label: 'Create Giveaway', id: 'CreateGiveaway' },
  { path: 'giveaways/active-giveaways', icon: 'fa-list', label: 'Active Giveaways', id: 'ActiveGiveaways' },
  { path: 'giveaways/premium', icon: 'fa-star', label: 'Premium', id: 'Premium' },
  { path: 'giveaways/blacklist', icon: 'fa-ban', label: 'Blacklist', id: 'Blacklist' },
];

const moderationRightNav = [
  { path: 'moderation/auto-mod', icon: 'fa-robot', label: 'Auto Mod', id: 'AutoMod' },
  { path: 'moderation/manual-actions', icon: 'fa-hand-paper', label: 'Manual Actions', id: 'ManualActions' },
  { path: 'moderation/reports', icon: 'fa-flag', label: 'Reports', id: 'Reports' },
  { path: 'moderation/logs', icon: 'fa-file-alt', label: 'Logs', id: 'Logs' },
];

const levelingRightNav = [
  { path: 'leveling/leveling', icon: 'fa-cog', label: 'Settings', id: 'Settings' },
  { path: 'leveling/leaderboard', icon: 'fa-trophy', label: 'Leaderboard', id: 'Leaderboard' },
  { path: 'leveling/rewards', icon: 'fa-gift', label: 'Rewards', id: 'Rewards' },
  { path: 'leveling/stats', icon: 'fa-chart-line', label: 'Stats', id: 'Stats' },
];

export function getRoute(hash) {
  const value = hash.startsWith('#/') ? hash.slice(2) : hash.startsWith('#') ? hash.slice(1) : hash;
  return value || 'dashboard';
}

export const routes = {
  dashboard: {
    componentName: 'Dashboard',
    pageTitle: 'Dashboard',
    sidebarActive: 'Dashboard',
    contentHtml: '',
    rightNav: [],
  },
  overview: {
    componentName: 'OverviewOverview',
    pageTitle: 'Overview',
    sidebarActive: 'Overview',
    contentHtml: '',
    rightNav: overviewRightNav,
  },
  'overview/statistics': {
    componentName: 'OverviewStatistics',
    pageTitle: 'Statistics',
    sidebarActive: 'Overview',
    contentHtml: '',
    rightNav: overviewRightNav,
  },
  'overview/settings': {
    componentName: 'OverviewSettings',
    pageTitle: 'Settings',
    sidebarActive: 'Overview',
    contentHtml: '',
    rightNav: overviewRightNav,
  },
  'overview/alerts': {
    componentName: 'OverviewAlerts',
    pageTitle: 'Alerts',
    sidebarActive: 'Overview',
    contentHtml: '',
    rightNav: overviewRightNav,
  },
  'overview/history': {
    componentName: 'OverviewHistory',
    pageTitle: 'History',
    sidebarActive: 'Overview',
    contentHtml: '',
    rightNav: overviewRightNav,
  },
  'servers/servers': {
    componentName: 'ServersServers',
    pageTitle: 'Servers',
    sidebarActive: 'Servers',
    contentHtml: '',
    rightNav: serversRightNav,
  },
  'servers/add-server': {
    componentName: 'ServersAdd_server',
    pageTitle: 'Add Server',
    sidebarActive: 'Servers',
    contentHtml: '',
    rightNav: serversRightNav,
  },
  'servers/backup': {
    componentName: 'ServersBackup',
    pageTitle: 'Backup',
    sidebarActive: 'Servers',
    contentHtml: '',
    rightNav: serversRightNav,
  },
  'servers/manage-roles': {
    componentName: 'ServersManage_roles',
    pageTitle: 'Manage Roles',
    sidebarActive: 'Servers',
    contentHtml: '',
    rightNav: serversRightNav,
  },
  'servers/permissions': {
    componentName: 'ServersPermissions',
    pageTitle: 'Permissions',
    sidebarActive: 'Servers',
    contentHtml: '',
    rightNav: serversRightNav,
  },
  'tickets/tickets': {
    componentName: 'TicketsTickets',
    pageTitle: 'Tickets',
    sidebarActive: 'Tickets',
    contentHtml: '',
    rightNav: ticketsRightNav,
  },
  'tickets/audit-logs': {
    componentName: 'TicketsAudit_logs',
    pageTitle: 'Audit Logs',
    sidebarActive: 'Tickets',
    contentHtml: '',
    rightNav: ticketsRightNav,
  },
  'tickets/forms': {
    componentName: 'TicketsForms',
    pageTitle: 'Forms',
    sidebarActive: 'Tickets',
    contentHtml: '',
    rightNav: ticketsRightNav,
  },
  'tickets/panels': {
    componentName: 'TicketsPanels',
    pageTitle: 'Panels',
    sidebarActive: 'Tickets',
    contentHtml: '',
    rightNav: ticketsRightNav,
  },
  'tickets/tags': {
    componentName: 'TicketsTags',
    pageTitle: 'Tags',
    sidebarActive: 'Tickets',
    contentHtml: '',
    rightNav: ticketsRightNav,
  },
  'tickets/ticket-embed': {
    componentName: 'TicketsTicket_embed',
    pageTitle: 'Ticket Embed',
    sidebarActive: 'Tickets',
    contentHtml: '',
    rightNav: ticketsRightNav,
  },
  'tickets/ticket-staff': {
    componentName: 'TicketsTicket_staff',
    pageTitle: 'Ticket Staff',
    sidebarActive: 'Tickets',
    contentHtml: '',
    rightNav: ticketsRightNav,
  },
  'tickets/transcripts': {
    componentName: 'TicketsTranscripts',
    pageTitle: 'Transcripts',
    sidebarActive: 'Tickets',
    contentHtml: '',
    rightNav: ticketsRightNav,
  },
  'giveaways/giveaway': {
    componentName: 'GiveawaysGiveaway',
    pageTitle: 'Giveaways',
    sidebarActive: 'Giveaways',
    contentHtml: '',
    rightNav: giveawaysRightNav,
  },
  'giveaways/active-giveaways': {
    componentName: 'GiveawaysActive_giveaways',
    pageTitle: 'Active Giveaways',
    sidebarActive: 'Giveaways',
    contentHtml: '',
    rightNav: giveawaysRightNav,
  },
  'giveaways/blacklist': {
    componentName: 'GiveawaysBlacklist',
    pageTitle: 'Blacklist',
    sidebarActive: 'Giveaways',
    contentHtml: '',
    rightNav: giveawaysRightNav,
  },
  'giveaways/create-giveaway': {
    componentName: 'GiveawaysCreate_giveaway',
    pageTitle: 'Create Giveaway',
    sidebarActive: 'Giveaways',
    contentHtml: '',
    rightNav: giveawaysRightNav,
  },
  'giveaways/premium': {
    componentName: 'GiveawaysPremium',
    pageTitle: 'Premium',
    sidebarActive: 'Giveaways',
    contentHtml: '',
    rightNav: giveawaysRightNav,
  },
  'moderation/moderation': {
    componentName: 'ModerationModeration',
    pageTitle: 'Moderation',
    sidebarActive: 'Moderation',
    contentHtml: '',
    rightNav: moderationRightNav,
  },
  'moderation/auto-mod': {
    componentName: 'ModerationAuto_mod',
    pageTitle: 'Auto Mod',
    sidebarActive: 'Moderation',
    contentHtml: '',
    rightNav: moderationRightNav,
  },
  'moderation/manual-actions': {
    componentName: 'ModerationManual_actions',
    pageTitle: 'Manual Actions',
    sidebarActive: 'Moderation',
    contentHtml: '',
    rightNav: moderationRightNav,
  },
  'moderation/reports': {
    componentName: 'ModerationReports',
    pageTitle: 'Reports',
    sidebarActive: 'Moderation',
    contentHtml: '',
    rightNav: moderationRightNav,
  },
  'moderation/logs': {
    componentName: 'ModerationLogs',
    pageTitle: 'Logs',
    sidebarActive: 'Moderation',
    contentHtml: '',
    rightNav: moderationRightNav,
  },
  'leveling/leveling': {
    componentName: 'LevelingLeveling',
    pageTitle: 'Leveling',
    sidebarActive: 'Leveling',
    contentHtml: '',
    rightNav: levelingRightNav,
  },
  'leveling/leaderboard': {
    componentName: 'LevelingLeaderboard',
    pageTitle: 'Leaderboard',
    sidebarActive: 'Leveling',
    contentHtml: '',
    rightNav: levelingRightNav,
  },
  'leveling/rewards': {
    componentName: 'LevelingRewards',
    pageTitle: 'Rewards',
    sidebarActive: 'Leveling',
    contentHtml: '',
    rightNav: levelingRightNav,
  },
  'leveling/stats': {
    componentName: 'LevelingStats',
    pageTitle: 'Stats',
    sidebarActive: 'Leveling',
    contentHtml: '',
    rightNav: levelingRightNav,
  },
};

File: api/cmd/dashctl/pages.go
package main

// pages.go is the source of truth for the dashboard's client-side routing
// table. `dashctl gen-pages` emits src/dashboard/data/pages.js from the
// definitions below.
//
// Adding or renaming a page: edit only this file, then run
//   dashctl gen-pages

// SidebarLink is one entry in the left rail.
type SidebarLink struct {
	Path  string // hash path, e.g. "tickets/tickets"
	Icon  string // FontAwesome class, e.g. "fa-ticket"
	Label string
	ID    string // matches Route.SidebarActive
}

// RightNavItem is one entry in a right-rail submenu.
type RightNavItem struct {
	Path  string
	Icon  string
	Label string
	ID    string
}

// Section bundles a sidebar tab with its submenu and routes.
type Section struct {
	JSVar    string         // JS variable holding the right-nav array
	Sidebar  SidebarLink    // entry inserted into sidebarLinks
	RightNav []RightNavItem // entries in the section's right rail
	Routes   []RouteEntry   // pages under this section
}

// RouteEntry maps a hash path to a Svelte component + page title.
type RouteEntry struct {
	Path          string
	Component     string
	PageTitle     string
	SidebarActive string // defaults to Section.Sidebar.ID
}

// rootRoutes lists routes that don't belong to a section's right-rail (e.g.
// the bare /dashboard landing page).
var rootRoutes = []RouteEntry{
	{Path: "dashboard", Component: "Dashboard", PageTitle: "Dashboard", SidebarActive: "Dashboard"},
}

// sections is the spine of the dashboard. Edit here, regen, done.
var sections = []Section{
	{
		JSVar: "overviewRightNav",
		Sidebar: SidebarLink{Path: "overview", Icon: "fa-chart-line", Label: "Overview", ID: "Overview"},
		RightNav: []RightNavItem{
			{Path: "overview/statistics", Icon: "fa-chart-bar", Label: "Statistics", ID: "Statistics"},
			{Path: "overview/settings", Icon: "fa-cog", Label: "Settings", ID: "Settings"},
			{Path: "overview/alerts", Icon: "fa-bell", Label: "Alerts", ID: "Alerts"},
			{Path: "overview/history", Icon: "fa-history", Label: "History", ID: "History"},
		},
		Routes: []RouteEntry{
			{Path: "overview", Component: "OverviewOverview", PageTitle: "Overview"},
			{Path: "overview/statistics", Component: "OverviewStatistics", PageTitle: "Statistics"},
			{Path: "overview/settings", Component: "OverviewSettings", PageTitle: "Settings"},
			{Path: "overview/alerts", Component: "OverviewAlerts", PageTitle: "Alerts"},
			{Path: "overview/history", Component: "OverviewHistory", PageTitle: "History"},
		},
	},
	{
		JSVar: "serversRightNav",
		Sidebar: SidebarLink{Path: "servers/servers", Icon: "fa-server", Label: "Servers", ID: "Servers"},
		RightNav: []RightNavItem{
			{Path: "servers/add-server", Icon: "fa-plus", Label: "Add Server", ID: "AddServer"},
			{Path: "servers/manage-roles", Icon: "fa-users-cog", Label: "Manage Roles", ID: "ManageRoles"},
			{Path: "servers/permissions", Icon: "fa-key", Label: "Permissions", ID: "Permissions"},
			{Path: "servers/backup", Icon: "fa-save", Label: "Backup", ID: "Backup"},
		},
		Routes: []RouteEntry{
			{Path: "servers/servers", Component: "ServersServers", PageTitle: "Servers"},
			{Path: "servers/add-server", Component: "ServersAdd_server", PageTitle: "Add Server"},
			{Path: "servers/manage-roles", Component: "ServersManage_roles", PageTitle: "Manage Roles"},
			{Path: "servers/permissions", Component: "ServersPermissions", PageTitle: "Permissions"},
			{Path: "servers/backup", Component: "ServersBackup", PageTitle: "Backup"},
		},
	},
	{
		JSVar: "ticketsRightNav",
		Sidebar: SidebarLink{Path: "tickets/tickets", Icon: "fa-ticket", Label: "Tickets", ID: "Tickets"},
		RightNav: []RightNavItem{
			{Path: "tickets/tickets", Icon: "fa-gears", Label: "Settings", ID: "Settings"},
			{Path: "tickets/categories", Icon: "fa-layer-group", Label: "Categories", ID: "Categories"},
			{Path: "tickets/panels", Icon: "fa-arrow-pointer", Label: "Panels", ID: "Panels"},
			{Path: "tickets/forms", Icon: "fa-file-lines", Label: "Forms", ID: "Forms"},
			{Path: "tickets/snippets", Icon: "fa-comment-dots", Label: "Snippets", ID: "Snippets"},
			{Path: "tickets/tags", Icon: "fa-tag", Label: "Tags", ID: "Tags"},
			{Path: "tickets/ticket-staff", Icon: "fa-people-group", Label: "Staff", ID: "TicketStaff"},
			{Path: "tickets/transcripts", Icon: "fa-file", Label: "Transcripts", ID: "Transcripts"},
			{Path: "tickets/statistics", Icon: "fa-chart-bar", Label: "Statistics", ID: "Statistics"},
			{Path: "tickets/audit-logs", Icon: "fa-file-medical", Label: "Audit logs", ID: "AuditLogs"},
		},
		Routes: []RouteEntry{
			{Path: "tickets/tickets", Component: "TicketsTickets", PageTitle: "Tickets"},
			{Path: "tickets/categories", Component: "TicketsCategories", PageTitle: "Categories"},
			{Path: "tickets/panels", Component: "TicketsPanels", PageTitle: "Panels"},
			{Path: "tickets/forms", Component: "TicketsForms", PageTitle: "Forms"},
			{Path: "tickets/snippets", Component: "TicketsSnippets", PageTitle: "Snippets"},
			{Path: "tickets/tags", Component: "TicketsTags", PageTitle: "Tags"},
			{Path: "tickets/ticket-embed", Component: "TicketsTicket_embed", PageTitle: "Ticket Embed"},
			{Path: "tickets/ticket-staff", Component: "TicketsTicket_staff", PageTitle: "Ticket Staff"},
			{Path: "tickets/transcripts", Component: "TicketsTranscripts", PageTitle: "Transcripts"},
			{Path: "tickets/statistics", Component: "TicketsStatistics", PageTitle: "Statistics"},
			{Path: "tickets/audit-logs", Component: "TicketsAudit_logs", PageTitle: "Audit Logs"},
		},
	},
	{
		JSVar: "giveawaysRightNav",
		Sidebar: SidebarLink{Path: "giveaways/giveaway", Icon: "fa-gift", Label: "Giveaways", ID: "Giveaways"},
		RightNav: []RightNavItem{
			{Path: "giveaways/create-giveaway", Icon: "fa-plus-circle", Label: "Create Giveaway", ID: "CreateGiveaway"},
			{Path: "giveaways/active-giveaways", Icon: "fa-list", Label: "Active Giveaways", ID: "ActiveGiveaways"},
			{Path: "giveaways/premium", Icon: "fa-star", Label: "Premium", ID: "Premium"},
			{Path: "giveaways/blacklist", Icon: "fa-ban", Label: "Blacklist", ID: "Blacklist"},
		},
		Routes: []RouteEntry{
			{Path: "giveaways/giveaway", Component: "GiveawaysGiveaway", PageTitle: "Giveaways"},
			{Path: "giveaways/active-giveaways", Component: "GiveawaysActive_giveaways", PageTitle: "Active Giveaways"},
			{Path: "giveaways/blacklist", Component: "GiveawaysBlacklist", PageTitle: "Blacklist"},
			{Path: "giveaways/create-giveaway", Component: "GiveawaysCreate_giveaway", PageTitle: "Create Giveaway"},
			{Path: "giveaways/premium", Component: "GiveawaysPremium", PageTitle: "Premium"},
		},
	},
	{
		JSVar: "moderationRightNav",
		Sidebar: SidebarLink{Path: "moderation/moderation", Icon: "fa-shield", Label: "Moderation", ID: "Moderation"},
		RightNav: []RightNavItem{
			{Path: "moderation/auto-mod", Icon: "fa-robot", Label: "Auto Mod", ID: "AutoMod"},
			{Path: "moderation/manual-actions", Icon: "fa-hand-paper", Label: "Manual Actions", ID: "ManualActions"},
			{Path: "moderation/reports", Icon: "fa-flag", Label: "Reports", ID: "Reports"},
			{Path: "moderation/logs", Icon: "fa-file-alt", Label: "Logs", ID: "Logs"},
		},
		Routes: []RouteEntry{
			{Path: "moderation/moderation", Component: "ModerationModeration", PageTitle: "Moderation"},
			{Path: "moderation/auto-mod", Component: "ModerationAuto_mod", PageTitle: "Auto Mod"},
			{Path: "moderation/manual-actions", Component: "ModerationManual_actions", PageTitle: "Manual Actions"},
			{Path: "moderation/reports", Component: "ModerationReports", PageTitle: "Reports"},
			{Path: "moderation/logs", Component: "ModerationLogs", PageTitle: "Logs"},
		},
	},
	{
		JSVar: "applicationsRightNav",
		Sidebar: SidebarLink{Path: "applications/applications", Icon: "fa-clipboard-check", Label: "Applications", ID: "Applications"},
		RightNav: []RightNavItem{
			{Path: "applications/applications", Icon: "fa-inbox", Label: "Review queue", ID: "ReviewQueue"},
			{Path: "applications/forms", Icon: "fa-list-check", Label: "Forms", ID: "Forms"},
			{Path: "applications/statistics", Icon: "fa-chart-bar", Label: "Statistics", ID: "Statistics"},
		},
		Routes: []RouteEntry{
			{Path: "applications/applications", Component: "ApplicationsApplications", PageTitle: "Applications"},
			{Path: "applications/forms", Component: "ApplicationsForms", PageTitle: "Application Forms"},
			{Path: "applications/statistics", Component: "ApplicationsStatistics", PageTitle: "Application Statistics"},
		},
	},
	{
		// Note: SidebarActive is "Dashboard" rather than the section ID below
		// because the legacy layout chose to highlight the gear icon.
		JSVar: "",
		Sidebar: SidebarLink{Path: "dashboard", Icon: "fa-gear", Label: "Settings", ID: "Dashboard"},
	},
	{
		JSVar: "levelingRightNav",
		Sidebar: SidebarLink{Path: "leveling/leveling", Icon: "fa-ranking-star", Label: "Leveling", ID: "Leveling"},
		RightNav: []RightNavItem{
			{Path: "leveling/leveling", Icon: "fa-cog", Label: "Settings", ID: "Settings"},
			{Path: "leveling/leaderboard", Icon: "fa-trophy", Label: "Leaderboard", ID: "Leaderboard"},
			{Path: "leveling/rewards", Icon: "fa-gift", Label: "Rewards", ID: "Rewards"},
			{Path: "leveling/stats", Icon: "fa-chart-line", Label: "Stats", ID: "Stats"},
		},
		Routes: []RouteEntry{
			{Path: "leveling/leveling", Component: "LevelingLeveling", PageTitle: "Leveling"},
			{Path: "leveling/leaderboard", Component: "LevelingLeaderboard", PageTitle: "Leaderboard"},
			{Path: "leveling/rewards", Component: "LevelingRewards", PageTitle: "Rewards"},
			{Path: "leveling/stats", Component: "LevelingStats", PageTitle: "Stats"},
		},
	},
}
